indexer/node/fetcher.go

```diff
@@ -0,0 +1,63 @@
+package node
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+const (
+	// Max number of headers that's bee returned by the Fetcher at once. This will
+	// eventually be configurable
+	maxHeaderBatchSize = 50
+)
+
+type Fetcher struct {
+	ethClient EthClient
+
+	// TODO: Store the last header block hash to ensure
+	// the next batch of headers builds on top
+	nextStartingBlockHeight *big.Int
+}
+
+// NewFetcher instantiates a new instance of Fetcher against the supplied rpc client.
+// The Fetcher will start retrieving blocks starting at `fromBlockHeight`.
+func NewFetcher(ethClient EthClient, fromBlockHeight *big.Int) (*Fetcher, error) {
+	fetcher := &Fetcher{
+		ethClient:               ethClient,
+		nextStartingBlockHeight: fromBlockHeight,
+	}
+
+	return fetcher, nil
+}
+
+// NextConfirmedHeaders retrives the next set of headers that have been
+// marked as finalized by the connected client
+func (f *Fetcher) NextFinalizedHeaders() ([]*types.Header, error) {
+	finalizedBlockHeight, err := f.ethClient.FinalizedBlockHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO:
+	//  - (unlikely) What do we do if our connected node is suddently behind by many blocks?
```

Comment:
I'd log a warning and return just like how you are

Reply:
sounds good